pages/login.js

```diff
@@ -69,7 +69,7 @@ class Login extends Component {
       }
       /* eslint-enable */
 
-      Router.push(searchParams.destination ? decodeURIComponent(searchParams.destination) : '/')
+      Router.pushRoute(searchParams.destination ? decodeURIComponent(searchParams.destination) : '/')
```

Comment:
Just a note that this is probably the most crucial part here.

`Router.pushRoute()` will automatically match the destination to a defined route if one exists. This means we don't need to worry about rewriting parameterized pages when they're encountered.